commit 7305acda01b1f4b66327cd2ce5e6b12acfb65de6
Author: Kefu Chai <kchai@redhat.com>
Date:   Thu Jul 16 19:39:13 2020 +0800

    mon: only dump non-empty "task status"
    
    * mgr/ServiceMap: add ServiceMap::Service::has_running_tasks() to tell
      if a service has any running tasks
    * mon/Monitor: do not print an empty "task status" section when printing
      cluster status. it's just confusing.
    
    Signed-off-by: Kefu Chai <kchai@redhat.com>

diff --git a/src/mgr/ServiceMap.cc b/src/mgr/ServiceMap.cc
index f2abcfb9ab8..490b118bcf0 100644
--- a/src/mgr/ServiceMap.cc
+++ b/src/mgr/ServiceMap.cc
@@ -91,6 +91,13 @@ std::string ServiceMap::Service::get_summary() const
   return ss.str();
 }
 
+bool ServiceMap::Service::has_running_tasks() const
+{
+  return std::any_of(daemons.begin(), daemons.end(), [](auto& daemon) {
+    return !daemon.second.task_status.empty();
+  });
+}
+
 std::string ServiceMap::Service::get_task_summary(const std::string_view task_prefix) const
 {
   // contruct a map similar to:
diff --git a/src/mgr/ServiceMap.h b/src/mgr/ServiceMap.h
index e91ce94ad15..ed027907c54 100644
--- a/src/mgr/ServiceMap.h
+++ b/src/mgr/ServiceMap.h
@@ -41,6 +41,7 @@ struct ServiceMap {
     static void generate_test_instances(std::list<Service*>& ls);
 
     std::string get_summary() const;
+    bool has_running_tasks() const;
     std::string get_task_summary(const std::string_view task_prefix) const;
     void count_metadata(const std::string& field,
 			std::map<std::string,int> *out) const;
